clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -696,8 +698,12 @@ public void handle(JoinGroupResponse joinResponse, RequestFuture<ByteBuffer> fut
     private RequestFuture<ByteBuffer> onJoinLeader(JoinGroupResponse joinResponse) {
         try {
             // perform the leader synchronization and send back the assignment for the group
-            Map<String, ByteBuffer> groupAssignment = performAssignment(joinResponse.data().leader(), joinResponse.data().protocolName(),
-                    joinResponse.data().members());
+            Map<String, ByteBuffer> groupAssignment = performAssignment(
+                joinResponse.data().leader(),
+                joinResponse.data().protocolName(),
+                joinResponse.data().members(),
+                joinResponse.data().skipAssignment()
```

Comment:
I think a comment about this would be helpful. An obvious question is why do we still call `performAssignment` when `skipAssignment` is set. It's useful to remember that we still need to propagate the leader and member state to the coordinator implementation.

Reply:
That makes sense. I have renamed the method to avoid the confusion and updated its javadoc.